### PR TITLE
Fix cell tracking dataset and simplify/unify data path specification in configs

### PR DESCRIPTION
**Pull request by shaikh58**
1) This PR adds support for training/testing models with data in the Cell Tracking Challenge format, a commonly used format in microscopy. It fixes the CellTrackingDataset class. In summary, CTC data is in the form of labelled masks. More info on the format is available here: https://celltrackingchallenge.net/datasets/

2) Currently, the data path specification can be confusing, with various ways of specifying data that interact with each others. This PR unifies this specification. Now, a directory path (or list of paths) must be provided, with suffix labels which are use to infer the type of dataset (Sleap or CTC). Support for the legacy method of specifying slp_files/video_files in lists has been removed.  Each directory in the list of directories may contain any number of slp/video file pairs. Note that the crop_size parameter must be of the same length as the directory path (dataset.train_dataset.dir.path) and specifies the crop size corresponding to the data in the directory at the same index

## Summary by CodeRabbit

- **New Features**
  - Introduced customizable dataset options with flexible directory configurations and the choice of tighter bounding boxes.
  - Added enhanced image processing utilities for accurate crop padding and bounding box extraction.
  
- **Bug Fixes**
  - Improved handling of crop sizes and bounding boxes to ensure consistent processing across datasets.

- **Refactor**
  - Streamlined data handling by adjusting dataset initialization and image chunk processing.
  - Removed legacy evaluation components to simplify the public interface.
  
- **Chores**
  - Removed unnecessary configurations related to the test dataset and updated training parameters for simplicity.

- **Tests**
  - Updated test configurations and assertions to align with the new dataset structures and a reduced training cycle.

### CODE REVIEW

File: dreem/io/config.py [hunk 1]
```diff
@@ -255,51 +247,83 @@ def get_dataset(
...
+            )
+        if self.labels_suffix is None or self.vid_suffix is None:
+            raise KeyError(
+                f"Must provide a labels suffix and vid suffix to search for but found {self.labels_suffix} and {self.vid_suffix}"
+            )
 
+        # TODO: handle this better!
+        if "model" in self.cfg:
+            if (
+                "encoder_type" in self.cfg["model"]["encoder_cfg"]
```
Comment (aaprasad) [inline dreem/io/config.py:266]: use `cfg.get` method!

File: dreem/io/config.py [hunk 2]
```diff
@@ -255,51 +247,83 @@ def get_dataset(
...
+                and self.cfg["model"]["encoder_cfg"]["encoder_type"] == "descriptor"
```
Comment (aaprasad) [inline dreem/io/config.py:267]: same as above

**Review by aaprasad — COMMENTED** [review]
use cfg.get method to prevent errors when keys don't exist

**Comment by shaikh58** [comment]
Had originally based this branch off the tight-box branch before it was merged. Once it was merged, this branch was rebased off main, hence the series of commits on Mar 31 that are duplicated from earlier.
